File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:


        hashMap = defaultdict(list)
        returnStatement = False

        for i in range(len(nums)):

            if nums[i] not in hashMap:
                hashMap[nums[i]].append(i)

            elif nums[i] in hashMap:
                hashMap[nums[i]].append(i)

                theList = hashMap[nums[i]]
                for j in range(len(theList)):

                
                    if (abs(theList[j]-i) <= k) and (theList[j]!=i):
                        returnStatement = True
                        return True


        return returnStatement



       




        

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
static const bool Booster = [](){
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map <int,int> myMap;
        for(int i=0;i<nums.size();i++){
            if(myMap.find(nums[i])!=myMap.end()){
                int tempJ = myMap[nums[i]];
                if ((i - tempJ) <= k) {
                    return true;
                }
            }
            myMap[nums[i]] = i;
        }
        return false;
    }
};